src/crypto/dleq.rs

```diff
@@ -0,0 +1,1048 @@
+use std::convert::TryInto;
+
+use crate::{
+    consensus::{self, CanonicalBytes},
+    crypto,
+};
+use amplify::num::u256;
+
+use bitcoin_hashes::{self, Hash};
+
+use bitvec::{order::Lsb0, prelude::BitSlice};
+use curve25519_dalek::{
+    constants::ED25519_BASEPOINT_POINT as G, edwards::CompressedEdwardsY as ed25519PointCompressed,
+    edwards::EdwardsPoint as ed25519Point, scalar::Scalar as ed25519Scalar,
+};
+
+const ENTROPY: bool = true;
+
+fn _max_ed25519() -> u256 {
+    (u256::from(1u32) << 252) + 27742317777372353535851937790883648493u128
+}
+
+fn reverse_endianness(bytes: &[u8; 32]) -> [u8; 32] {
+    let mut bytes_rev = *bytes;
+    bytes_rev.reverse();
+    bytes_rev
+}
+
+// matches https://github.com/monero-project/monero/blob/9414194b1e47730843e4dbbd4214bf72d3540cf9/src/ringct/rctTypes.h#L454
+// i.e. hash-to-curve of G as in https://github.com/monero-project/mininero/blob/master/mininero.py#L305-L323
+// TODO: this is disgusting and must be removed asap
```

Comment:
Can this TODO be removed now?

Reply:
No - I'll address that in the followup PR. Disgusting := inefficient.